allennlp/training/trainer.py

```diff
@@ -7,6 +7,7 @@
 import traceback
 from typing import Dict, List, Optional, Tuple, Union, Any
 
+from apex import amp
```

Comment:
I would suggest this:

```
try:
	from apex import amp
except ImportError:
	amp = None

```

and then below, check that `amp is not None` and raise a `ConfigurationError` if it is and the user has passed an opt level. Does that make sense?

Reply:
Done.